pkgs/development/python-modules/apsw/default.nix

```diff
@@ -1,21 +1,29 @@
-{ stdenv, buildPythonPackage, fetchFromGitHub
+{ stdenv, buildPythonPackage, fetchFromGitHub, fetchpatch
 , sqlite, isPyPy }:
 
 buildPythonPackage rec {
   pname = "apsw";
-  version = "3.22.0-r1";
+  version = "3.29.0-r1";
 
   disabled = isPyPy;
 
   src = fetchFromGitHub {
     owner = "rogerbinns";
     repo = "apsw";
     rev = version;
-    sha256 = "02ldvshcgr4c7c8anp4flfnw8g8ys5bflkb8b51rb618qxhhwyak";
+    sha256 = "1p3sgmk9qwd0a634lirva44qgpyq0a74r9d70wxb6hsa52yjj9xb";
   };
 
   buildInputs = [ sqlite ];
 
+  patches = [
+    # Fixes a test failure with sqlite 3.30, see https://github.com/rogerbinns/apsw/issues/275
+    (fetchpatch {
+      url = "https://github.com/rogerbinns/apsw/commit/13df0b57bff59542978abf7c0a440c9274e3aac3.diff";
+      sha256 = "17rlp9xv1xilvb830i3iwypzg52nfylw4b8xx97jzb1c2wcpxz8k";
```

Comment:
```
hash mismatch in fixed-output derivation '/nix/store/jm798wz9r03z77lgyhbbvixlfhy81j44-13df0b57bff59542978abf7c0a440c9274e3aac3.diff':
  wanted: sha256:17rlp9xv1xilvb830i3iwypzg52nfylw4b8xx97jzb1c2wcpxz8k
  got:    sha256:1wi1mfis2mr21389wdnvq44phg0bpm5vpwmxhvrj211vwfm0q7dv
```

Reply:
Hm, odd. Lemme double-check.

Reply:
Well, indeed. Thanks! I wonder why that sometimes happens. I certainly have not added the file manually to the store with the wrong hash, but I did use git-prefetch-url, which seems to sometimes not work as expected.

Reply:
that's why i always just change the last character, so that it's invalid and forces a cache miss :)